lib/assert.js

```diff
@@ -283,6 +298,26 @@ function _deepEqual(actual, expected, strict, memos) {
   return objEquiv(actual, expected, strict, memos);
 }
 
+function setEquiv(a, b) {
+  // This behaviour will work for any sets with contents that have
+  // strict-equality. That is, it will return false if the two sets contain
+  // equivalent objects that aren't reference-equal. We could support that, but
+  // it would require scanning each pairwise set of not strict-equal items,
+  // which is O(n^2), and would get exponentially worse if sets are nested. So
+  // for now we simply won't support deep equality checking set items or map
+  // keys.
```

Comment:
For tests, I agree. I'm nervous people might be using `assert.deepEqual` inside runtime code, although I just checked the `node_modules` directory of a project I'm working on with ~400 transitive dependancies and only `jsprim` calls assert.deepEqual outside of a test, and thats a in a very minor way. I'll wait for some more feedback, but unless there's any strong objections I'll change the behaviour to be correct and slow.